lib/index.ts

```diff
@@ -0,0 +1,40 @@
+import * as bodyParser from 'body-parser';
+import * as express from 'express';
+import * as _ from 'lodash';
+
+import {NetworkManager} from './nm';
+import wifiApi from './routes/wifi';
+
+interface Route {
+	method: string;
+	handler: (nm: NetworkManager) => (req: any, res: any) => void;
+}
+
+function start(port: number = 1337) {
+	const app = express();
+	app.use(bodyParser.json());
+	const nm = new NetworkManager();
+
+	return nm.init()
+	.then(() => {
+		const exposeNmApiToApp = exposeAPIs(app)(nm);
+		exposeNmApiToApp('/wifi')(wifiApi);
+		app.listen(port);
+	})
+	.catch((err) => {
+		throw err;
+	});
+}
+
+function exposeAPIs(app) {
+	return (nm: any) => (rootApi: string = '/') => (APIs: any) => {
+		const root: string = (rootApi === '/') ? '' : rootApi;
+		_.each(APIs, (route: Route, path: string) => {
+			return app[route.method.toLowerCase()](`${root}/${path}`, route.handler(nm));
```

Comment:
Should this maybe use `path.posix.join(root, routePath)` instead of string concat?

Reply:
I guess it could... Why `posix` though?

Reply:
Because it'd use backslashes if run on Windows and using `path.join()` (see https://nodejs.org/api/path.html#path_path_posix), which would break

Reply:
But thinking about it; it's probably best to user a Router, and use that as middleware with a mount path, e.g.
```js
const router = new Express.Router()

_.each(APIs, (route, routePath) => {
  router[route.method](routePath, route.handler(nm))
})

app.use( `/${root}`, router )
```

Reply:
Wait what? Does Windows actually support `DBus`/`NetworkManager`?